src/main/java/seedu/address/model/EduMateHistory.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+
+/**
+ * Wraps past commands entered by user.
+ */
+public class EduMateHistory implements ReadOnlyEduMateHistory {
+
+    private ArrayList<String> eduMateHistory;
+    private int index = 0;
+    private final int minIndex = 0;
+    private final int maxIndex = 100;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        eduMateHistory = new ArrayList<>();
+    }
+
+    public EduMateHistory() {}
+
+    /**
+     * Creates an EduMateHistory using the ArrayList in the {@code toBeCopied}.
+     */
+    public EduMateHistory(ReadOnlyEduMateHistory toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    /**
+     * Replaces the contents of the command history list with {@code history}.
+     */
+    public void setList(ArrayList<String> history) {
+        this.eduMateHistory = history;
+    }
+
+    /**
+     * Replaces the exisitng contents of the current data of this {@code EduMateHistory} with {@code newData}.
+     */
+    public void resetData(ReadOnlyEduMateHistory newData) {
+        requireNonNull(newData);
+
+        setList(newData.getHistory());
+    }
+
+    /**
+     * Adds a command to the current command history.
+     * @param command Command string
+     */
+    public void addCommand(String command) {
+        assert command != null;
```

Comment:
Use `requiresNonNull()`, but I applaud your effort to employ Defensive Programming. Though, I would have preferred `assert(command != null)` with the brackets